src/components/app/list-new/HelmAppList.tsx

```diff
@@ -61,92 +62,89 @@ import {
+}: HelmAppListProps) {
     const [dataStateType, setDataStateType] = useState(AppListViewType.LOADING)
     const [errorResponseCode, setErrorResponseCode] = useState(0)
     const [devtronInstalledHelmAppsList, setDevtronInstalledHelmAppsList] = useState<HelmApp[]>([])
     const [externalHelmAppsList, setExternalHelmAppsList] = useState<HelmApp[]>([])
-    const [filteredHelmAppsList, setFilteredHelmAppsList] = useState<HelmApp[]>([])
-    const [sortBy, setSortBy] = useState(SortBy.APP_NAME)
-    const [sortOrder, setSortOrder] = useState(OrderBy.ASC)
-    const [clusterIdsCsv, setClusterIdsCsv] = useState('')
-    const [appStatus, setAppStatus] = useState(_getAppStatusFromRequestUrl())
     const [sseConnection, setSseConnection] = useState<EventSource>(undefined)
     const [externalHelmListFetchErrors, setExternalHelmListFetchErrors] = useState<string[]>([])
     const [showGuidedContentCards, setShowGuidedContentCards] = useState(false)
-    const location = useLocation()
-    const history = useHistory()
-    const params = new URLSearchParams(location.search)
 
-    // component load
-    useEffect(() => {
-        init()
-    }, [])
+    const { appStatus, environment, cluster, namespace, project, searchKey, sortBy, sortOrder, offset, pageSize } = filterConfig
 
-    // it means filter/sorting has been applied
-    useEffect(() => {
-        if (dataStateType === AppListViewType.LIST) {
-            if (clusterIdsCsv === _getClusterIdsFromRequestUrl() && appStatus === _getAppStatusFromRequestUrl()) {
-                handleFilteration()
-            } else {
-                init()
-            }
+    const getFilteredAppList = (): HelmApp[] => {
+        let tempFilteredList: HelmApp[] = [...devtronInstalledHelmAppsList, ...externalHelmAppsList]
+        if (searchKey) {
+            tempFilteredList = tempFilteredList.filter((app) => app.appName.includes(searchKey.toLowerCase()))
         }
-    }, [payloadParsedFromUrl])
-
-    // on data rendering first time
-    useEffect(() => {
-        if (dataStateType == AppListViewType.LIST) {
-            handleFilteration()
+        if (project?.length) {
+            tempFilteredList = tempFilteredList.filter((app) => project.includes(String(app.projectId)))
         }
-    }, [dataStateType])
+        if (environment?.length) {
+            tempFilteredList = tempFilteredList.filter((app) =>
+                environment.includes(String(app.environmentDetail.environmentId)),
+            )
+        }
+        if (namespace?.length) {
+            tempFilteredList = tempFilteredList.filter((app) =>
+                namespace.includes(`${app.environmentDetail.clusterId}_${app.environmentDetail.namespace}`),
+            )
+        }
+        if (appStatus?.length) {
+            tempFilteredList = tempFilteredList.filter((app) => appStatus.includes(app.appStatus))
+        }
+
+        return tempFilteredList
+    }
+
+    const filteredHelmAppsList: HelmApp[] = getFilteredAppList()
```

Comment:
Can memoize and return sorted, sliced filtered array and also return total count from that method